eos-link-host-mali-driver

```diff
@@ -22,14 +22,75 @@
-MALI_DIR=/usr/lib/mali400/arm-linux-gnueabihf
 
-DEST_DIR="/var/lib/flatpak/extension/$EXTENSION_REF"
-mkdir -p $(dirname "$DEST_DIR")
-ln -sf "$MALI_DIR" "$DEST_DIR"
+#
+# create_flatpak_gl_host_extension FLATPAK_DIR MALI_DRIVER
+#
+# Creates a flatpak GL host extension in $FLATPAK_DIR from $MALI_DRIVER
+#
+function create_flatpak_gl_host_extension {
+    FLATPAK_DIR=$1
+    MALI_DRIVER=$2
+
+    if test ! -d $FLATPAK_DIR; then
+        return;
+    fi
+
+    DEST_DIR=$FLATPAK_DIR/extension/$EXTENSION_REF
+
+    # Ensure destination directory
+    mkdir -p $DEST_DIR
+
+    # Create lib link for flatpak GL extension
+    ln -sr $DEST_DIR $DEST_DIR/lib
 
-if test -d /var/endless-extra/flatpak; then
-    DEST_DIR="/var/endless-extra/flatpak/extension/$EXTENSION_REF"
-    mkdir -p $(dirname "$DEST_DIR")
-    ln -sf "$MALI_DIR" "$DEST_DIR"
+    # Hardlink mali driver
+    ln -t $DEST_DIR $MALI_DRIVER
+
+    # Create mali driver wrapper with missing symbol
+    # (Binary generated from libMaliWrapper.c)
+(   base64 -d <<EOF
+H4sIANofxVkAA5VVTWwbRRR+s14nm8RN05CSpA2RSxXJbZIlIeYnVELE2Wz6k1RpC1KEiuy1vWIN
+6x/trlFDQC0SBw5Ar0UcGoRUcQNEuKWA1KiqEIdE6iGHVDJVTQ8cnByq+Gbe7Myq28VFYqRPb968
+N2/evr+9PDOnEkLAWyGIAeViIYA40u/CAIII4ThEQUTZPojAY222FJEh7HLElbMz4mIF+TuIFi4X
+AvIryF/hcjyGc1UnC00WlXe5PjLaidiP6GimzFeY26R3JEQboh3Ryu1FEPuecpfGgMWEef5AJJxv
+gy/p3TDju5C/7JNT/m2fHerrFwH9UwH5SIC3A/a+CtyPB/TNJ/Ql+Cag3x/Qp5JP8PvWkXbT71tU
+cnbJ1JYu6M6MZRWtk1oha+oWlBzD0rVsMl929EtJs5h5H5A3cRMQlQuucDFoIMOUzVx6XjNzsl2E
+pJ7VHA2SybRtJ21Hsxz/PpnknF7IuntOkaCvG/iNz2JS71KKSdyiFBO7TSkWQoVSLAyxMloVK9er
+nZWrFWnnl+o2rfHbV+93/f75/e3dtaq3v+vb/+Hbr/v2a7h/5szO6rIyaIDcfe8dY9i4lhj4Nbwr
+CXBU3Nxfeas+8cZKIrJBJnrVnxICnj/alM4sK18bJ37cWeubfUFdNmD40Qbd96k7a6XZVfTnU0T9
+51b12Fyn+pEi7kagsXe7lnkPhl/eGJr/WBkxDANGXr0XMq7ja8BfG6ws1CcWvk8cxtcOqN/y1w6d
+vjjwodKT+hNtXKv14duCupgSd+ECDJETwt+3SKI+UTqbEipEnTfOprpOXlIiY429z2oxtX6zE7nG
+3gc1cjSTXkn99pDER1IQ29qMqtKRxl62JqhGauvhIvqLwwJ6EBLiL6yi5xqNBq0ljffZKu859NXt
+GomwfrsBrE+9/l0X2RyIIXoQBxC9iB+EpzTjfyxFfDyRtoR/Y3Z6+rVobIaWo21H4/Kk/OLo+Fg5
+XS445UmdHcfTen78GBPCFO0uTdfSOSDH6Qe9MjoFLe2tUxJpE9rFbrGHHCS9oT7STw4Jh0MDZFB4
+noBsG7ZjOVoaZEOzsVKySwV7Kc+oY4Fs6aZcMh2QsV0c91jL5zIgv1tEDise5Ewxn9cLyE2dn5c1
+x7Fyaewt+3/EooPHX+C5oHiT88BzJPJ4d/BzmiMKOhRCPD8iz1U/5109geGG7z0v8keAzdYEsBxS
+xPg9bwbT+TvE77T4crXgs+f5d9ynR+cwxQB5Uo+uMa5DfWT/E/YvCfn06LuT3J5Xe+soiDZ593Wu
+T/XoPH0gsu8IvjvN7Y1xnuqNB+JCMcdslbzzgxiMl5rEz/PNW6dR72ITvX8ASrgeTLAHAAA=
+EOF
+) | gunzip > $DEST_DIR/libEGL.so.1
+
+    chmod +x $DEST_DIR/libEGL.so.1
+
+    # Create extra EGL and GLES hard links
+    for LIB in libEGL.so.1.4 libGLESv2.so.2 libGLESv2.so.2.0; do
+      ln $DEST_DIR/libEGL.so.1 $DEST_DIR/$LIB
+      chmod +x $DEST_DIR/$LIB
+    done
+}
+
+# Lookup mali driver
+if test -f /usr/lib/mali400/arm-linux-gnueabihf/libMali.so; then
+    MALI_DRIVER="/usr/lib/mali400/arm-linux-gnueabihf/libMali.so"
+elif test -f /usr/lib/arm-linux-gnueabihf/mali400-egl-x11/libMali.so; then
```

Comment:
We are assuming the new path on master on the unit file that runs this script, so this will never be true. I believe it is safe and simpler to assume the new location on master, but that means you'll need to adjust the path on the fix for eos3.2. The link should be updated on upgrade (before this change).